---
title: "Functional groups"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
documentclass: book
biblio-style: apalike
link-citations: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = F, error=F, message = F)
#options(repos = BiocManager::repositories())
pacman::p_load('dplyr', 'tidyr', 'htmltools', 'bbplot', 'scales',
               'ggplot2', 'rdrop2', 'shiny', 'BiocManager',
               'dendextend', 'data.table', 'Biostrings', 'alakazam', "unikn", 
               'plotly', 'ggdendro', "RColorBrewer", install = F)
```

```{r}
source("functions.R")
```

## Sequence pre-processing

Summary statistics for each of the functional groups declared in the app.

The app includes the P1 and P11 naive datasets and the P4 non-naive dataset.

For P1 and P11 the following filtration criteria were applied:

* Functional sequence, no stop codons or frame shifts.
* Sequences which start from position 1 of the V gene.
* Sequences which didn't have gaps open (-) and didn't include any N's
* After changing into group annotations, sequences which had more than a single assignment in naive repertoires were remove.

The groups were created with similarity of 95% based on complete linkage and functional sequences and up to position 318.

## Functional groups

```{r, fig.asp=0.8, out.width="1000%", fig.width=20}
load("data.rda")
mat <- mat_list$IGH$functional$nonsingle$all$`318`
clust <- hclust(as.dist(mat), "complete")
clusters <- cutree(clust, h = 0.05)
names(clusters) <- names(cutree(clust, h = 0.05))
dend <- as.dendrogram(clust)
fam_clusters <- clusters

clusters <- clusters[order.dendrogram(dend)]
clusters_numbers <- unique(clusters) - (0 %in% clusters)
n_clusters <- length(clusters_numbers)

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

mypal = setNames(alpha(pal %>% usecol(n = n_clusters),0.99), 1:n_clusters) 
cluster_new <- mypal[clusters]

clusters_imgt <- clusters
clusters_imgt <- getGene(names(clusters_imgt), strip_d = F, omit_nl = F)

imgt_colors <-
  setNames(alpha(pal %>% usecol(n = length(unique(clusters_imgt))), 0.99), sort(unique(clusters_imgt)))
clusters_imgt <-   imgt_colors[clusters_imgt]

dend2 <-
  dend %>% branches_attr_by_clusters(clusters, values = mypal) %>% dendextend::set("labels_cex", 0.9)


par(mar = c(12, 4, 1, 1))
dend2 %>% plot
dend2 %>% rect.dendrogram2(
  k = n_clusters,lower_rect = 0,
  text = clusters_numbers, xpd = T,
  #border = 8,
  lty = 5,
  lwd = 2,
  border = mypal
)
colored_bars(
  cbind(clusters_imgt, cluster_new),
  dend2,
  sort_by_labels_order = F,
  rowLabels = c("IMGT", "NEW")
)
```


## Projects sequence depth

```{r}
load("data_frac_new2.rda")

data <- data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`

depth <- data %>% dplyr::group_by(project, subject) %>% dplyr::summarise(counts = sum(count))

ggplot(depth, aes(counts, fill = project)) + geom_histogram() + facet_grid(.~project) + 
  bbplot::bbc_style() + scale_color_manual(values = alpha(c("#1380A1", "#FAAB18", "#990000"),0.99)) +
  theme(axis.text.x = element_text(size = 12, angle = 45), 
        axis.text.y = element_text(size = 14), legend.position = "none") #+
  	#scale_x_continuous(labels = scales::label_number(suffix = " K", scale = 1e-5)) # millions

```


<!--chapter:end:index.Rmd-->

# "IGHV1-18"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% rowwise() %>% mutate(gene = getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV1-18G1"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-18.docx", local_path = "docs/IGHV1-18.docx",
#               overwrite = TRUE, verbose = F)
# #drop_df <- textreadr::read_docx("docs/IGHV1-18.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-18.docx", to = "markdown", output = "IGHV1-18_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/p7wwxwxsgtk2j425oi1jx/IGHV1-18.docx?dl=0&rlkey=m6dwbte5avwqmdlfiwjs973t5')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       htmlOutput("df_output")
#     )
# )
# 
# server <- function(input, output) {
#   
#   observeEvent(input$ab2, {
#       drop_download("public/conclusions/IGHV1-18.docx", local_path = "docs/IGHV1-18.docx",
#                 overwrite = TRUE, verbose = F)
#       #drop_df <- textreadr::read_docx("docs/IGHV1-18.docx")
#       
#       rmarkdown::pandoc_convert(input = "IGHV1-18.docx", to = "markdown", output = "IGHV1-18_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), verbose = F)
#       output$markdown <- renderUI({
#         withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-18_docx.md', quiet = TRUE)))
#     })
#     
#   })
#   
#   output$markdown <- renderUI({
#     includeMarkdown(knitr::knit('docs/IGHV1-18_docx.md', quiet = TRUE))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

From the results we belive that the cutoff for this group should be closer to $12.5\%$, and for the adjusted states the allele combinations and the relations are stated in the table below.

```{r}

tableData <- data_cutoff(data, func_groups, g_group, 12.5, or_allele) %>% dplyr::group_by(zygousity_state, v_allele) %>% dplyr::summarise(mean_freq = paste0(round(quantile(freq, 3/4),3),":", round(quantile(freq, 1/4),3)), v_alleles_abc = unique(v_alleles_abc)) %>% dplyr::group_by(zygousity_state, v_alleles_abc) %>% dplyr::summarise(fractions = paste0(mean_freq, collapse = ";"))

DT::datatable(
        tableData,
        options = list(dom = "tipr"),
        selection = 'none',
        colnames = c(
          "Zygousity state" = "zygousity_state",
          "V\nallele" = "v_alleles_abc",
          "IQR range" = "fractions"
        )
      )


# column names change to Zygousity state, alleles combinations, IQR 

```

[this site](https://peresay.shinyapps.io/cluster_app/?url=%22page_allele%22&func_group=%22IGHV1-2%22&allele_thresh=%212.52%22)


```{r}

tmp_allele_db <-
  allele_db %>% filter(grepl(func_groups[as.character(g)], new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(func_groups[as.character(g)], "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 12.5

tmp <- data_cutoff(data, func_groups, g_group, allele_thresh, or_allele)

```

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

<!--chapter:end:G1.rmd-->

# "IGHV1-8"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV1-8G10"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G10.rmd-->

# "IGHV2-26"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV2-26G11"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G11.rmd-->

# "IGHV2-5"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV2-5G12"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G12.rmd-->

# "IGHV2-70/IGHV2-70D"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV2-70G13"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G13.rmd-->

# "IGHV3-11"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV3-11G14"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G14.rmd-->

# "IGHV3-13"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV3-13G15"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G15.rmd-->

# "IGHV3-15"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV3-15G16"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G16.rmd-->

# "IGHV3-20"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV3-20G17"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G17.rmd-->

# "IGHV3-21"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV3-21G18"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G18.rmd-->

# "IGHV3-23/IGHV3-23D"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV3-23G19"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G19.rmd-->

# "IGHV1-2"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV1-2G2"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

From the results we belive that the cutoff for this group should be closer to $2\%$ except allele **05** that should stay at $0.5\%$                    , and for the adjusted states the allele combinations and the relations are stated in the table below.

### Allele specific cutoff: $2\%$ for all except $0.5\%$ for allele **05**
```{r}

tmp_allele_db <-
  allele_db %>% filter(grepl(func_groups[as.character(g)], new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(func_groups[as.character(g)], "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = setNames(rep(2,length(names(or_allele))),names(or_allele))
allele_thresh[5]<-0.5

tmp <- data_cutoff_allele_thresh(data, func_groups, g_group, allele_thresh, or_allele)

```

```{r}

tableData <- tmp %>% dplyr::group_by(zygousity_state, v_allele) %>% dplyr::summarise(mean_freq = paste0(round(quantile(freq, 3/4),3),":", round(quantile(freq, 1/4),3)), v_alleles_abc = unique(v_alleles_abc)) %>% dplyr::group_by(zygousity_state, v_alleles_abc) %>% dplyr::summarise(fractions = paste0(mean_freq, collapse = ";"))

DT::datatable(
        tableData,
        options = list(dom = "tipr"),
        selection = 'none',
        colnames = c(
          "Zygousity state" = "zygousity_state",
          "V\nallele" = "v_alleles_abc",
          "IQR range" = "fractions"
        )
      )

```

For interactive play on the threshold and relative allele fractions, please refer to [this site](https://peresay.shinyapps.io/cluster_app/?url=%22page_allele%22&func_group=%22IGHV1-2%22&allele_thresh=%222%22)

```{r, warning = F, error=F, message = F}
p_list <- c()
allele_thresh = setNames(rep(2,length(names(or_allele))),(or_allele))
allele_thresh[5]<-0.5

for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  
  p_list[[i]] <- plot_zygousity_allele_thresh(tmp, i, allele_thresh, g) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

### Allele specific cutoff: $5\%$ for all except $0.5\%$ for allele **05**

```{r}

tmp_allele_db <-
  allele_db %>% filter(grepl(func_groups[as.character(g)], new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(func_groups[as.character(g)], "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = setNames(rep(5,length(names(or_allele))),names(or_allele))
allele_thresh[5]<-0.5

tmp <- data_cutoff_allele_thresh(data, func_groups, g_group, allele_thresh, or_allele)

```

```{r}

tableData <- tmp %>% dplyr::group_by(zygousity_state, v_allele) %>% dplyr::summarise(mean_freq = paste0(round(quantile(freq, 3/4),3),":", round(quantile(freq, 1/4),3)), v_alleles_abc = unique(v_alleles_abc)) %>% dplyr::group_by(zygousity_state, v_alleles_abc) %>% dplyr::summarise(fractions = paste0(mean_freq, collapse = ";"))

DT::datatable(
        tableData,
        options = list(dom = "tipr"),
        selection = 'none',
        colnames = c(
          "Zygousity state" = "zygousity_state",
          "V\nallele" = "v_alleles_abc",
          "IQR range" = "fractions"
        )
      )

```

For interactive play on the threshold and relative allele fractions, please refer to [this site](https://peresay.shinyapps.io/cluster_app/?url=%22page_allele%22&func_group=%22IGHV1-2%22&allele_thresh=%222%22)

```{r, warning = F, error=F, message = F}
p_list <- c()
allele_thresh = setNames(rep(5,length(names(or_allele))),(or_allele))
allele_thresh[5]<-0.5

for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  
  p_list[[i]] <- plot_zygousity_allele_thresh(tmp, i, allele_thresh, g) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

<!--chapter:end:G2.rmd-->

# "IGHV3-30/IGHV3-30-3/IGHV3-30-5/IGHV3-33"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV3-30G20"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G20.rmd-->

# "IGHV3-35"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV3-35G21"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## Observations

For this group we've seen a single allele, hence the general summary statistics do not apply here and were omitted.

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G21.rmd-->

# "IGHV3-43/IGHV3-43D"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV3-43G22"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G22.rmd-->

# "IGHV3-48"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV3-48G23"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G23.rmd-->

# "IGHV3-49"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV3-49G24"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G24.rmd-->

# "IGHV3-53/IGHV3-66"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV3-53G25"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G25.rmd-->

# "IGHV3-62"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV3-62G26"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

No rearrangments were found for this group, hence the general summary statistics do not apply here and were omitted.
## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G26.rmd-->

# "IGHV3-64/IGHV3-64D"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV3-64G27"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G27.rmd-->

# "IGHV3-7"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV3-7G28"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G28.rmd-->

# "IGHV3-72"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV3-72G29"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## Observations

For this group we've seen a single allele, hence the general summary statistics do not apply here and were omitted.

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G29.rmd-->

# "IGHV1-24"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV1-24G3"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## Observations

For this group we've seen a single allele, hence the general summary statistics do not apply here and were omitted.

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G3.rmd-->

# "IGHV3-73"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV3-73G30"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G30.rmd-->

# "IGHV3-74"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV3-74G31"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G31.rmd-->

# "IGHV3-9"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV3-9G32"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G32.rmd-->

# "IGHV4-28"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV4-28G33"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G33.rmd-->

# "IGHV4-30-2/IGHV4-30-4"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV4-30-2G34"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G34.rmd-->

# "IGHV4-30-4/IGHV4-31"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV4-30-4G35"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G35.rmd-->

# "IGHV4-34"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV4-34G36"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G36.rmd-->

# "IGHV4-34"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV4-34G37"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G37.rmd-->

# "IGHV4-34"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV4-34G38"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## Observations

For this group we've seen a single allele, hence the general summary statistics do not apply here and were omitted.

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G38.rmd-->

# "IGHV4-38-2/IGHV4-39"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV4-38-2G39"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G39.rmd-->

# "IGHV1-3"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV1-3G4"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G4.rmd-->

# "IGHV4-4"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV4-4G40"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G40.rmd-->

# "IGHV4-4/IGHV4-59"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV4-4G41"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G41.rmd-->

# "IGHV4-59"

```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV4-59G42"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

No rearrangments were found for this group, hence the general summary statistics do not apply here and were omitted.

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G42.rmd-->

# "IGHV4-59"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV4-59G43"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## Observations

For this group we've seen a single allele, hence the general summary statistics do not apply here and were omitted.

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G43.rmd-->

# "IGHV4-61"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV4-61G44"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G44.rmd-->

# "IGHV5-10-1"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV5-10-1G45"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G45.rmd-->

# "IGHV5-51"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV5-51G46"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G46.rmd-->

# "IGHV6-1"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV6-1G47"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G47.rmd-->

# "IGHV7-4-1"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV7-4-1G48"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G48.rmd-->

# "IGHV1-45"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV1-45G5"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G5.rmd-->

# "IGHV1-46"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV1-46G6"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G6.rmd-->

# "IGHV1-58"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV1-58G7"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G7.rmd-->

# "IGHV1-69/IGHV1-69D"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV1-69G8"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Group alignment

Based on the viewd alleles, we calculated the distance between the germline sequences.

```{r, fig.width=18,fig.height=30}
v_calls <- unique(data[grepl(g_group,v_gene),v_call])
seq_align(v_calls, allele_db, vgerms, chain, mat, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## 0.5% cutoff 

We set an initial cutoff of $0.5\%$ to determine the potential genotype priors. For this cutoff we examined the zygousity state, such as homozygousity, heterozygousity and so on.


```{r}

tmp_allele_db <-
  allele_db %>% dplyr::filter(grepl(as.character(g_group), new_allele)) %>%
  dplyr::group_by(new_allele) %>% dplyr::summarise(or_allele = paste0(or_allele, collapse = "/"))

or_allele <-
  setNames(gsub(chain, "", as.character(tmp_allele_db$or_allele)), as.character(gsub(
    paste0(g_group, "[*]"),
    "",
    tmp_allele_db$new_allele
  )))

allele_thresh = 0.5

tmp <- data_cutoff(data, func_g_groups, g_group, allele_thresh, or_allele)

```

With the selected cutoff we saw that there are `r length(unique(tmp %>% arrange(zygousity_state) %>% pull(zygousity_state)))` zygousity states.

```{r, warning = F, error=F, message = F}
p_list <- c()
for(i in as.character(unique(tmp %>% dplyr::arrange(zygousity_state) %>% dplyr::pull(zygousity_state)))){
  p_list[[i]] <- plot_zygousity(tmp, i, allele_thresh, g_group) %>% config(edits = list(shapePosition = TRUE),
                    scrollZoom = FALSE) %>% layout(title = paste0("state ", i))
  #
}

tagList(p_list)
```

## Observations

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G8.rmd-->

# "IGHV1-69-2"
```{r}
source("functions.R")
```

```{r, echo=FALSE}
load("data.rda")

chain = "IGH"
func <-  data.frame(allele = names(vgerms[[chain]]), functionality = !grepl("(ORF|P)",sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4])), sign = sapply(seqinr::getAnnot(vgerms_full[[chain]]), function(x) unlist(strsplit(x,"[|]"))[4]), stringsAsFactors = F)

mat <- mat_list$IGH$functional$nonsingle$all$`318`

load("data_frac_new2.rda")
data <- setDT(data_frac$IGH$functional$nonsingle$all$`318`$complete$`95`)
data[,v_call:=paste0(v_gene,"*",v_allele)]
load("alleles_dbs.rda")
allele_db <- alleles_dbs$IGH$functional$nonsingle$all$`318`$complete$`95`
allele_db <- allele_db %>% dplyr::rowwise() %>% dplyr::mutate(gene = alakazam::getGene(or_allele, strip_d = F, omit_nl = F), group = strsplit(gsub(gene, "", new_allele),"[*]")[[1]][1], gene_group = alakazam::getGene(new_allele, strip_d = F, omit_nl = F))
load("functional_groups.rda")
func_groups <- functional_groups$IGH$functional$nonsingle$all$`318`$complete$`95`

cols <- c("#FAAB18", "#1380A1","#990000", "#588300")

pal <- cols %>% 
  newpal(names = c("orangy", "bluish", "redish", "greeny"))

edit_links <- readLines("edit_links.txt")
share_links <- readLines("share_links.txt")
```

```{r,echo=FALSE}
g_group = "IGHV1-69-2G9"
group = names(func_groups)[func_groups==g_group]
gr <- allele_db %>% filter(gene_group == g_group) %>% pull(group) %>% unique()
g <- allele_db %>% filter(gene_group == g_group) %>% pull(gene) %>% unique()
```

## Allele appearnce 

The group of `r group` includes `r sum(grepl(paste0(g,"[*]", collapse = "|"), func$allele))` alleles, `r sum(func$functionality[grepl(paste0(g,"[*]", collapse = "|"), func$allele)])` out of the alleles are functional.

For each allele we counted the number of appearances across the population, any appearance was considered valid. 

```{r}
allele_appearance(data, g_group)
```

## Sequence depth

To examine the potential cutoff we observed the sequence depth for each allele

```{r}
sequence_depth(data, g_group)
```

## Observations

For this group we've seen a single allele, hence the general summary statistics do not apply here and were omitted.

This section is editable by clicking on the edit button below. To refresh the section click on the refresh button

You can access the file also from [here](`r edit_links[grep(g_group,edit_links)]`)

```{r, message=F, echo=F, warning=F}
knitr::include_url(paste0("https://peresay.shinyapps.io/G2_group/?group=",g_group))
# # Authenticate and save token for later use2
# token <- drop_auth(rdstoken = "dropbox_token.rds")
# 
# # Retrieveing your file is as simple as
# drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = FALSE, progress = FALSE)
# drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
# invisible(rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = FALSE))
# 
# ui <- fluidPage(
# 
#     # Application title
#     mainPanel(
#       shiny::actionButton(inputId='ab1', label="Edit text", 
#                           icon = icon("edit"), 
#                           onclick ="window.open('https://www.dropbox.com/scl/fi/crphl3u49fmkw8wnb5v71/IGHV1-2.docx?dl=0&rlkey=l54oqhrlxx8wyp6bywr224sx0')"),
#       
#       shiny::actionButton(inputId='ab2', label="Reload text", 
#                           icon = icon("sync-alt")),
#       
#       #htmlOutput("df_output"),
#       uiOutput('markdown')
#     )
# )
# 
# server <- function(input, output) {
#     
#   observeEvent(input$ab2, {
#     drop_download("public/conclusions/IGHV1-2.docx", local_path = "docs/IGHV1-2.docx",
#               overwrite = TRUE, verbose = F)
#     #drop_df <- textreadr::read_docx("docs/IGHV1-2.docx")
#     
#     rmarkdown::pandoc_convert(input = "IGHV1-2.docx", to = "markdown", output = "IGHV1-2_docx.md", options = c("--wrap=none","--reference-links","--metadata-file=metadata.yaml","--standalone"), wd = "docs", verbose = F)
#     output$markdown <- renderUI({
#       #withMathJax(includeMarkdown(knitr::knit('IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
#     
#   })
#     
#   # output$df_output <- renderUI({
#   #       if(input$ab2) return()
#   #       HTML(paste(drop_df, collapse = "<br/><br/>"))
#   #   })
#   
#   output$markdown <- renderUI({
#     #withMathJax(includeMarkdown(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#     HTML(markdown::markdownToHTML(knitr::knit('docs/IGHV1-2_docx.md', quiet = TRUE)))
#   })
# }
# 
# shinyApp(ui = ui, server = server)

```

## Conclusions

<!--chapter:end:G9.rmd-->

